libs/gi/ui/src/components/team/TeamCard.tsx

```diff
@@ -357,15 +411,236 @@ function CharacterArea({
-              )}
+              )} */}
             </Box>
           </Box>
         </Suspense>
       </DataContext.Provider>
     </CharacterContext.Provider>
   )
 }
+function WeaponCard({ weapon }: { weapon: ICachedWeapon }) {
+  return (
+    <CardThemed
+      bgt="neutral600"
+      sx={{
+        height: '100%',
+        maxHeight: '50px',
+        display: 'flex',
+        flexDirection: 'horizontal',
+        boxShadow: `0 0 10px rgba(0,0,0,0.4)`,
+      }}
+    >
+      <Box
+        component="img"
+        src={weaponAsset(weapon.key, weapon.ascension >= 2)}
+        maxWidth="100%"
+        maxHeight="100%"
+        sx={{ mt: 'auto' }}
+      />
+      <Box
+        sx={{
+          pr: 1,
+          display: 'flex',
+          flexDirection: 'column',
+          justifyContent: 'space-evenly',
+          color: 'neutral200.main',
+        }}
+      >
+        <Typography sx={{}}>
+          {getLevelString(weapon.level, weapon.ascension)}
+        </Typography>
+
+        {weaponHasRefinement(weapon.key) && (
+          <Typography>R{weapon.refinement}</Typography>
+        )}
+      </Box>
+    </CardThemed>
+  )
+}
+
+function ArtifactCard({ artifactData }: { artifactData: ArtifactData }) {
+  const { setNum = {}, mains = {} } = artifactData
+  const { t } = useTranslation('statKey_gen')
+  const processedSetNum = Object.entries(setNum).filter(
+    ([, num]) => num === 2 || num === 4
+  )
+  return (
+    <CardThemed
+      bgt="neutral600"
+      sx={{
+        height: '100%',
+        maxHeight: '50px',
+        display: 'flex',
+        flexDirection: 'horizontal',
+        boxShadow: `0 0 10px rgba(0,0,0,0.4)`,
+        flexGrow: 1,
+      }}
+    >
+      <Box
+        sx={{
+          Width: '50px',
+          minWidth: '50px',
+          height: '50px',
+          position: 'relative',
+        }}
+      >
+        {processedSetNum.length === 2 ? (
+          <Set22 sets={processedSetNum.map(([set]) => set)} />
+        ) : processedSetNum.length === 1 ? (
+          <Set4 set={processedSetNum[0][0]} num={processedSetNum[0][1]} />
+        ) : (
+          false
+        )}
+      </Box>
+      <Box
+        sx={{
+          display: 'flex',
+          flexGrow: 1,
+          position: 'relative',
+          justifyContent: 'space-around',
+          alignItems: 'center',
+        }}
+      >
+        {Object.entries(mains)
+          .filter(([, statKey]) => statKey)
+          .map(([slotKey, statKey]) => (
+            <BootstrapTooltip
+              key={slotKey + statKey}
+              title={
+                <Box>
+                  <Typography>
+                    <ArtifactSlotName slotKey="sands" />
+                  </Typography>
+                  <Typography>{t(statKey)}</Typography>
+                </Box>
+              }
+            >
+              <Typography sx={{ lineHeight: 0 }}>
+                {slotKey === 'sands' && (
+                  <SandsIcon sx={{ fontSize: 'inherit' }} />
```

Comment:
I hovered over this and crashed

Please enter the steps to reproduce this error below:
1. 

**Don't forget to upload your JSON file!**

```
Message: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
Name: Error
Stack: Error: Minified React error #185; visit https://reactjs.org/docs/error-decoder.html?invariant=185 for the full message or use the non-minified dev environment for full errors and additional helpful warnings.
    at du (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:33862)
    at efe (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:46:22379)
    at https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:131:23586
    at Ty (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:24296)
    at Yx (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:31641)
    at IV (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:31497)
    at IV (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:31405)
    at IV (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:31405)
    at IV (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:31405)
    at IV (https://genshin-optimizer-prs.github.io/pr/2323/frontend/assets/index-C6KbjYFe.js:48:31405)
```